### PR TITLE
MAINT Remove unused parametrize in `test_figure_rst_srcset`

### PR DESCRIPTION
**Pull request by lucyleeow**
Also removes unused parametrization in `test_figure_rst_srcset`, and move private funcs to above their use.

Originally `visit_srcset_other` was here to check adding other builders see: https://github.com/sphinx-gallery/sphinx-gallery/pull/808#issuecomment-818034757, but not used so removed here.

### CODE REVIEW

File: sphinx_gallery/directives.py [hunk 1]
```diff
@@ -190,6 +176,28 @@ def _parse_srcset(st):
     return srcset
 
 
+def _copy_images(self, node):
```
Comment (larsoner) [inline sphinx_gallery/directives.py:179]: Can you move this back down to minimize the `diff`? It helps with `git blame`

Comment (lucyleeow) [reply sphinx_gallery/directives.py:179]: Yeah diff is ugly, I'll revert.

**Comment by lucyleeow** [comment]
I think the removal of unused functions was done in #1169  , I might have done that by mistake since lint doesn't pass if we don't remove.

**Comment by larsoner** [comment]
Thanks @lucyleeow !

**Pull request merged**